### PR TITLE
JAMES-3400 Update documentation

### PR DESCRIPTION
**Pull request by quantranhong1999**

### CODE REVIEW

File: server/protocols/webadmin-cli/README.md [hunk 1]
```diff
@@ -123,4 +151,171 @@ Show all users' name on the list.
...
+```
+
+Resource name usernameToBeUsed should be an existing user 
+
+Resource name mailboxNameToBeTested should not be empty
+
+### Delete a mailbox and its children
+
+```
+{cli} mailbox delete <usernameToBeUsed> --mailbox <mailboxNameToBeDeleted>
```
Comment (chibenwa) [inline server/protocols/webadmin-cli/README.md:184]: Why do we need --mailbox here?

File: server/protocols/webadmin-cli/README.md [hunk 2]
```diff
@@ -123,4 +151,171 @@ Show all users' name on the list.
...
+Show all configured domain mappings
+```
+{cli} domain-mapping list
+```
+
+### Listing all destination domains for a source domain
+
+With sourceDomain.tld as the value passed to fromDomain resource name, this command will show all destination domains configured to that domain
+```
+{cli} domain-mapping list --sourcedomain <sourceDomain.tld>
```
Comment (chibenwa) [inline server/protocols/webadmin-cli/README.md:246]: Why --sourcedomain only here ?

File: server/protocols/webadmin-cli/README.md (hunk 1 above)
Comment (quantranhong1999) [reply server/protocols/webadmin-cli/README.md:184]: I make --mailbox option to put "Delete a mailbox and its children" and "Delete all mailboxes of a user" in 1 command line.

--mailbox option is not required. If there is --mailbox option, will delete only a specified mailbox; else delete all mailboxes of a user

File: server/protocols/webadmin-cli/README.md (hunk 2 above)
Comment (quantranhong1999) [reply server/protocols/webadmin-cli/README.md:246]: Same as above, to put "'Show all configured domain mappings" and "Listing all destination domains for a source domain" in 1 command

File: server/protocols/webadmin-cli/README.md (hunk 1 above)
Comment (chibenwa) [reply server/protocols/webadmin-cli/README.md:184]: How about ...

```
{cli} mailbox delete <usernameToBeUsed> --mailbox <mailboxNameToBeDeleted>
{cli} mailbox delete <usernameToBeUsed> --allMailboxes
```

?

File: server/protocols/webadmin-cli/README.md [hunk 3]
```diff
@@ -123,4 +151,171 @@ Show all users' name on the list.
...
+```
+{cli} mailbox list <usernameToBeUsed>
+```
+
+Resource name usernameToBeUsed should be an existing user
+
+### Reindex a user mailboxes
+
+```
+{cli} mailbox reindex --username <usernameToBeUsed>
```
Comment (rouazana) [inline server/protocols/webadmin-cli/README.md:210]: why here do you have a `--username <username>` option while in other commands you have only `<username>`?

Comment (rouazana) [reply server/protocols/webadmin-cli/README.md:210]: (can we try to be me more coherent?)

File: server/protocols/webadmin-cli/README.md [hunk 4]
```diff
@@ -123,4 +151,171 @@ Show all users' name on the list.
...
+Show all configured domain mappings
+```
+{cli} domain-mapping list
+```
+
+### Listing all destination domains for a source domain
+
+With sourceDomain.tld as the value passed to fromDomain resource name, this command will show all destination domains configured to that domain
+```
+{cli} domain-mapping list --sourcedomain <sourceDomain.tld>
```
Comment (chibenwa) [inline server/protocols/webadmin-cli/README.md:246]: Can we find a way to remove this --sourcedomain ?

**Review by chibenwa — APPROVED** [review]

**Review by rouazana — COMMENTED** [review]
I'm not sure this is related to this PR, but there are still some incoherence, for example:

{cli} mailbox exist <usernameToBeUsed> <mailboxNameToBeTested>
but
{cli} mailbox delete <usernameToBeUsed> --mailbox <mailboxNameToBeDeleted>

{cli} domain-mapping list --sourcedomain <sourceDomain.tld>
but
{cli} domain-mapping create <sourceDomain.tld> <destination.tld>

also
{cli} mailbox delete <usernameToBeUsed> --allMailboxes
is not clear because when you do a mailbox command, you don't expect that the main parameter will be a user...

so question for the team: what about always specifying the kind of option you expect?
mailbox list --user <user>
mailbox list -- user <user> --all

maybe only for create methods it does not make sense... I'm not sure. Time to read a good guide about writing a CLI (and avoiding some horrible nightmare like git ;))

(and don't hesitate to handle these ideas in an other PR)

**Review by rouazana — COMMENTED** [review]
it seems I was not reading the last version of the file, last version is really better, thanks!

**Comment by chibenwa** [comment]
> it seems I was not reading the last version of the file, last version is really better, thanks!

Yes, we worked together on this ;-)

**Comment by quantranhong1999** [comment]
Updated jwt options configuration

File: server/protocols/webadmin-cli/README.md [hunk 5]
```diff
@@ -30,7 +30,25 @@ $ ./james-cli --url http://127.0.0.1:9999 domain list
 ```
 
 The above command lists all domain names available on domain route at address http://127.0.0.1:9999. 
-It does not require any argument to execute. Options --url are optional. Without it, the default value is http://127.0.0.1:8000.
+It does not require any argument to execute. 
+
+#### Options are optional:
```
Comment (chibenwa) [inline server/protocols/webadmin-cli/README.md:35]: Can we be less extravagant on the title?

File: server/protocols/webadmin-cli/README.md [hunk 6]
```diff
@@ -32,7 +32,7 @@ $ ./james-cli --url http://127.0.0.1:9999 domain list
 The above command lists all domain names available on domain route at address http://127.0.0.1:9999. 
 It does not require any argument to execute. 
 
-#### Options are optional:
+Options are optional:
```
Comment (chibenwa) [inline server/protocols/webadmin-cli/README.md:35]: I suggest `The following options can be used:`

as options are necessarily optionals, overwise we would not call them options...

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**